pattern_test.go

```diff
@@ -41,6 +41,61 @@ func TestPatternErrorHandling(t *testing.T) {
 	}
 }
 
+// test that adding an empty pattern doesn't screw anything up
+func TestEmptyValueArray(t *testing.T) {
+	var err error
+	empties := []string{
+		`{"data": {"field": []}}`,
+		`{"data": {"field": [], "field2": [23]}}`,
+		`{"data": {"field2": [], "field": [23]}}`,
+	}
+	wanted := map[string][]X{
+		`{"data": {"field": 23}}`:  {empties[2]},
+		`{"data": {"field2": 23}}`: {empties[1]},
+		`{"data": {"field": []}}`:  {},
+	}
+	cm := newCoreMatcher()
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("addPattern panicked")
+		}
+	}()
+	for _, empty := range empties {
+		err = cm.addPattern(empty, empty)
+		if err != nil {
+			t.Error("addPattern: " + err.Error())
+		}
+	}
+
+	for want := range wanted {
+		matches, err := cm.matchesForJSONEvent([]byte(want))
+		if err != nil {
+			t.Errorf("m4je err on %s", want)
+		}
+		checkXEqual(t, wanted[want], matches)
+	}
+}
+func checkXEqual(t *testing.T, x1s []X, x2s []X) {
```

Comment:
nit: I just wonder whether `slices.Sort` and then `slices.Equal` from https://pkg.go.dev/golang.org/x/exp/slices would be more clear instead of creating a new equality check helper?

Reply:
Good idea. Unfortunately the "X" type doesn't implement cmp.ordered.

Reply:
Updated PR